spec/core/query_spec.rb

```diff
@@ -267,6 +267,16 @@
           ).to eq(answer)
       end
 
+      it "handles empty data" do
+        answer = Daru::DataFrame.new({
+          number: [nil],
+          sym: [:seven],
+          names: [Float::NAN.to_s]
+          }, index: Daru::Index.new([6])
+        )
+        expect(@df.where(@df[:sym].eq(:seven))).to eq(answer)
```

Comment:
There is an old saying "Don't believe test you've never seen failing". It means: you should first make sure this test is failing without your patch, then apply the patch, then check test is green. 
Hint: it doesn't check anything (on current daru master it will work).